Sources/StreamChat/Workers/MessageUpdater.swift

```diff
@@ -198,6 +198,35 @@ class MessageUpdater: Worker {
             }
         }
     }
+
+    func loadReactions(
+        cid: ChannelId,
+        messageId: MessageId,
+        pagination: Pagination,
+        completion: ((Result<MessageReactionsPayload, Error>) -> Void)? = nil
+    ) {
+        let endpoint: Endpoint<MessageReactionsPayload> = .loadReactions(
+            messageId: messageId,
+            pagination: pagination
+        )
+
+        apiClient.request(endpoint: endpoint) { result in
+            switch result {
+            case let .success(payload):
+                self.database.write({ session in
+                    try payload.reactions.forEach { try session.saveReaction(payload: $0) }
```

Comment:
do we actually want to store reactions coming from pagination in the database? sounds like an easy way to create bugs + it does not add a lot of value

Reply:
what I mean is, paginating reactions is a very rare use-case and its totally fine to hit the API every time (while slower the advantage is that you avoid potential sync bugs)

Reply:
With offline storage turned off we actually store everything in memory 🙂 The advantage of having reactions saved to DB is that when another message controller is instantiated it will expose the same list of reactions as the one used to paginate them initially.

We had a similar discussion about message search, right? Following existing patterns has both pros and cons and persisting some info indeed doesn't bring much value. But on the other hand, keeping it consistent before we come up with the new pattern that will apply to all knows cases kinda makes sense.

Reply:
Yes, I agree there's no value in saving these to the DB. This is the pattern followed across the codebase, so that is why it is done like this. Changing this without properly planning the Offline Refactoring it is a bit risky